packages/common/src/project/readers/ipfs-reader.ts

```diff
@@ -28,28 +31,26 @@ export class IPFSReader implements Reader {
     return undefined;
   }
 
-  async getPkg(): Promise<IPackageJson | undefined> {
-    return Promise.resolve(undefined);
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async getPkg(): Promise<IPackageJson> {
+    throw new Error('Project package.json is not found');
   }
 
-  async getProjectSchema(): Promise<unknown | undefined> {
+  async getProjectSchema(): Promise<unknown> {
     const projectYaml = await this.getFile(this.cid);
-    if (projectYaml === undefined) {
-      throw new Error(`Failed to fetch project from IPFS: ${this.cid}`);
-    }
     return yaml.load(projectYaml);
   }
 
-  async getFile(fileName: string): Promise<string | undefined> {
+  async getFile(fileName: string): Promise<string> {
     try {
       if (this.cache[fileName] === undefined) {
         const resolvedFileName = fileName.replace('ipfs://', '');
         this.cache[fileName] = this.resultToBuffer(this.ipfs.cat(resolvedFileName));
       }
       return await this.cache[fileName];
     } catch (e) {
-      console.error(`Failed to fetch file from IPFS: ${fileName}`, e);
-      return undefined;
+      console.error(`Failed to fetch project from IPFS: ${fileName}`, e);
+      throw new Error(`Failed to fetch project from IPFS: ${fileName}`);
```

Comment:
We have started providing the cause to all errors where we can. It would be worth reading this https://medium.com/with-orus/the-5-commandments-of-clean-error-handling-in-typescript-93a9cbdf1af5

```suggestion
      throw new Error(`Failed to fetch project from IPFS: ${fileName}`, { cause: e });
```